process/block/shardblock.go

```diff
@@ -1862,13 +1846,17 @@ func (sp *shardProcessor) DecodeBlockHeader(dta []byte) data.HeaderHandler {
 
```

Comment:
should make a function like createNewHeader - which returns a data.HeaderHandler implementation.

Reply:
Done